axum-core/src/ext_traits/request_parts.rs

```diff
@@ -54,7 +52,7 @@ pub trait RequestPartsExt: sealed::Sealed + Sized {
     ///     }
     /// }
     /// ```
-    fn extract<E>(&mut self) -> BoxFuture<'_, Result<E, E::Rejection>>
+    fn extract<E>(&mut self) -> impl Future<Output = Result<E, E::Rejection>> + Send + '_
```

Comment:
Is the `+ '_` needed? I think the impl trait return type should capture the `&mut self` lifetime implicitly.